core/os/os_linux.odin

```diff
@@ -893,8 +894,10 @@ get_env :: proc(key: string, allocator := context.allocator) -> (value: string)
 
 set_env :: proc(key, value: string) -> Errno {
 	runtime.DEFAULT_TEMP_ALLOCATOR_TEMP_GUARD()
-	s := strings.concatenate({key, "=", value, "\x00"}, context.temp_allocator)
-	res := _unix_putenv(strings.unsafe_string_to_cstring(s))
+  key_cstring := strings.unsafe_string_to_cstring(strings.concatenate({key, "\x00"}, context.temp_allocator))
```

Comment:
I would replace these incantations with `strings.clone_to_cstring(str, context.temp_allocator)` instead of hacking together a nul-terminated string and then casting it to a `cstring`.

Reply:
Once this is addressed it's fine to merge as far as I'm concerned.

Reply:
Fixed in f048ad13b52d3c2be2fcd975f1039eb88604360f

Reply:
Good catch about the putenv thing.